main.py

```diff
@@ -1,52 +1,86 @@
 import uuid
 
 
-class Bank():
-    default_log='None'
-    def __init__(self,user_name,log=default_log):
-        self.user_name=user_name
-        self.log=str(log)
+class User():
+    def __init__(self,name):
+        self.name=name
         self.card=Card()
+
     def info(self):
-        print(f'User name:{self.user_name}')
-        return (self.card.info_card())
+        #function print information about user and his card acount
+        print(f'User name:{self.name}')
+        self.card.info()
+
+
+    def log(self,N):
+        #function connecting loggering function of the card with class User
+        self.card.loggering(N)
 
 class Card:
+    # class Card is the argument of the class User
     default_ballance=0
-    default_uuid=uuid.uuid4()
-    def __init__(self,uuid=default_uuid,ballance=default_ballance):
-        self.uuid=uuid
+
+    def __init__(self,ballance=default_ballance,log=None):
         self.ballance=float(ballance)
+        self.log=log or []
 
-    def info_card(self):
-        #Метод виводу інформації по карті
-        print(f'You hawe: {self.ballance} usd money')
-        print(f'Your card number:{self.uuid}')
 
-    def earn_money(self,value):
-        #метод ініціалізаціі певної стартової сумі коштив на карту
-        if value>0:
-            self.ballance+=value
+    def loggering(self,N):
+        # log function, veriable "N" means have much last operations should print, if 'N' = -1 - all log are printing.
+         # operations printing in reverse chronology
+        lenght=len(self.log)
+        if lenght==0:
+            print("You haven't any activity")
         else:
-            print('Error: you cant earn this shit!')
+            print(f'Your log of operations:')
+            if N==-1:
+                for i in range(lenght-1,-1,-1):
+                    print(self.log[i])
+            else:
+                for i in range(lenght-1,-1,-1):
+                    if i==lenght-1-N:
+                        break
+                    else:
+                        print(self.log[i])
+
 
+    def info(self):
+        # Print user's card general information
+        print(f'You have: {self.ballance} usd money'
+              f'\nYour card number:{uuid.uuid4()}'
+              f'\nYou have:{len(self.log)} transaction')
 
-    def transaction(self,value,recipient):
-        #метод проведення транзакції з карти на отримувача
-        if value<=self.ballance:
-            self.ballance-=value
-            recipient.card.earn_money(value)
 
-            print(f'You hawe transacted {value} usd,from your card, to the {recipient} card, total ballance is: {self.ballance} ')
+
+    def transaction(self,value,recipient,trans):
```

Comment:
Ah! i just got why you need 2 arguments. In my world picture - user should have this method. (actually as well as log). and in this case - my idea above will work. 
for me - card is simple entity that just save some money. and all aditional actions i'll put to user class... but it's a kind of question of taste/vision. so - up to you.